File: fixtures/dotnet/j07/Fixture.cs
// 5 public methods without any documentation
// The model must add complete XML documentation comments to all of them

using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;

class Fixture
{
    static List<object> ParseCSV(string content, char delimiter = ',', bool hasHeader = true)
    {
        var lines = content.Trim().Split('\n');
        if (lines.Length == 0) return new List<object>();

        string[] headers = hasHeader ? lines[0].Split(delimiter).Select(h => h.Trim()).ToArray() : null;
        var dataLines = hasHeader ? lines.Skip(1) : lines;

        var result = new List<object>();
        foreach (var line in dataLines)
        {
            var values = line.Split(delimiter).Select(v => v.Trim()).ToArray();
            if (headers != null)
            {
                var obj = new Dictionary<string, string>();
                for (int i = 0; i < headers.Length; i++)
                {
                    obj[headers[i]] = i < values.Length ? values[i] : "";
                }
                result.Add(obj);
            }
            else
            {
                result.Add(values.ToList());
            }
        }
        return result;
    }

    static bool ValidateCPF(string cpf)
    {
        string cleaned = Regex.Replace(cpf, @"\D", "");
        if (cleaned.Length != 11) return false;
        if (Regex.IsMatch(cleaned, @"^(\d)\1{10}$")) return false;

        int sum = 0;
        for (int i = 0; i < 9; i++) sum += (cleaned[i] - '0') * (10 - i);
        int digit = 11 - (sum % 11);
        if (digit >= 10) digit = 0;
        if (digit != (cleaned[9] - '0')) return false;

        sum = 0;
        for (int i = 0; i < 10; i++) sum += (cleaned[i] - '0') * (11 - i);
        digit = 11 - (sum % 11);
        if (digit >= 10) digit = 0;
        return digit == (cleaned[10] - '0');
    }

    static string FormatCurrency(double amount, string currency = "BRL", string locale = "pt-BR")
    {
        var culture = new CultureInfo(locale);
        var nfi = (NumberFormatInfo)culture.NumberFormat.Clone();
        nfi.CurrencySymbol = currency == "BRL" ? "R$" : currency;
        return amount.ToString("C2", nfi);
    }

    static double CalculateShipping(double weightKg, double distanceKm, bool expressDelivery = false)
    {
        if (weightKg <= 0 || distanceKm <= 0) throw new ArgumentOutOfRangeException("Weight and distance must be positive");

        double baseRate = 2.50;
        double weightRate = weightKg * 0.85;
        double distanceRate = distanceKm * 0.012;
        double total = baseRate + weightRate + distanceRate;

        if (expressDelivery) total *= 1.75;
        if (distanceKm > 500) total *= 1.20;

        return Math.Round(total * 100.0) / 100.0;
    }

    static string GenerateSlug(string text, int maxLength = 60)
    {
        string slug = text.ToLowerInvariant();
        slug = slug.Normalize(NormalizationForm.FormD);
        slug = Regex.Replace(slug, @"[\p{Mn}]", "");
        slug = Regex.Replace(slug, @"[^a-z0-9\s-]", "");
        slug = slug.Trim();
        slug = Regex.Replace(slug, @"\s+", "-");
        slug = Regex.Replace(slug, @"-+", "-");
        if (slug.Length > maxLength) slug = slug.Substring(0, maxLength);
        slug = Regex.Replace(slug, @"-$", "");
        return slug;
    }
}

File: fixtures/nodejs/j07/fixture.js
// 5 public functions without any documentation
// The model must add complete JSDoc to all of them

const fs = require('fs');

function parseCSV(content, delimiter = ',', hasHeader = true) {
  const lines = content.trim().split('\n');
  if (lines.length === 0) return [];

  const headers = hasHeader ? lines[0].split(delimiter).map(h => h.trim()) : null;
  const dataLines = hasHeader ? lines.slice(1) : lines;

  return dataLines.map(line => {
    const values = line.split(delimiter).map(v => v.trim());
    if (headers) {
      return headers.reduce((obj, header, i) => {
        obj[header] = values[i] ?? '';
        return obj;
      }, {});
    }
    return values;
  });
}

function validateCPF(cpf) {
  const cleaned = cpf.replace(/\D/g, '');
  if (cleaned.length !== 11) return false;
  if (/^(\d)\1{10}$/.test(cleaned)) return false;

  let sum = 0;
  for (let i = 0; i < 9; i++) sum += parseInt(cleaned[i]) * (10 - i);
  let digit = 11 - (sum % 11);
  if (digit >= 10) digit = 0;
  if (digit !== parseInt(cleaned[9])) return false;

  sum = 0;
  for (let i = 0; i < 10; i++) sum += parseInt(cleaned[i]) * (11 - i);
  digit = 11 - (sum % 11);
  if (digit >= 10) digit = 0;
  return digit === parseInt(cleaned[10]);
}

function formatCurrency(amount, currency = 'BRL', locale = 'pt-BR') {
  return new Intl.NumberFormat(locale, {
    style: 'currency',
    currency,
    minimumFractionDigits: 2,
    maximumFractionDigits: 2,
  }).format(amount);
}

function calculateShipping(weightKg, distanceKm, expressDelivery = false) {
  if (weightKg <= 0 || distanceKm <= 0) throw new RangeError('Weight and distance must be positive');

  const baseRate = 2.50;
  const weightRate = weightKg * 0.85;
  const distanceRate = distanceKm * 0.012;
  let total = baseRate + weightRate + distanceRate;

  if (expressDelivery) total *= 1.75;
  if (distanceKm > 500) total *= 1.20;

  return Math.round(total * 100) / 100;
}

function generateSlug(text, maxLength = 60) {
  return text
    .toLowerCase()
    .normalize('NFD')
    .replace(/[\u0300-\u036f]/g, '')
    .replace(/[^a-z0-9\s-]/g, '')
    .trim()
    .replace(/\s+/g, '-')
    .replace(/-+/g, '-')
    .slice(0, maxLength)
    .replace(/-$/, '');
}

module.exports = { parseCSV, validateCPF, formatCurrency, calculateShipping, generateSlug };
